### PR TITLE
Add topKCategoricalAccuracy && sparseTopKCategoricalAccuracy metrics

### PR DESCRIPTION
**Pull request by syt123450**
FEATURE

This PR add topKCategoricalAccuracy metrics(feature requested in [tensorflow/tfjs#27](https://github.com/tensorflow/tfjs/issues/27)) and sparseTopKCategoricalAccuracy metrics(feature requested in [tensorflow/tfjs#26](https://github.com/tensorflow/tfjs/issues/26)).

This PR:
* Add topKCategoricalAccuracy metric and sparseTopKCategoricalAccuracy metric
* Add relative tests
* Register new metrics to [src/exports_metrics.ts](https://github.com/tensorflow/tfjs-layers/blob/master/src/exports_metrics.ts)

Note:
* this PR depends on [tensorflow/tfjs-core#1734](https://github.com/tensorflow/tfjs-core/pull/1734)

Reference:
* [Implementation of top_k_categorical_accuracy and sparse_top_k_categorical_accuracy in Keras](https://github.com/keras-team/keras/blob/master/keras/metrics.py#L46)
* [top_k_categorical_accuracy and sparse_top_k_categorical_accuracy test cases in Keras](https://github.com/keras-team/keras/blob/master/tests/keras/metrics_test.py#L85)

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/537)

### CODE REVIEW

**Comment by caisq** [comment]
@syt123450 Can you please fix the test / lint errors:

> ERROR: build step 0 "node:10" failed: exit status 1
> ERROR
> Finished Step #0 - "yarn"
> Step #0 - "yarn": info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> Step #0 - "yarn": error Command failed with exit code 1.
> Step #0 - "yarn": info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> Step #0 - "yarn": error Command failed with exit code 2.
> Step #0 - "yarn": 
> Step #0 - "yarn": Found 2 errors.
> Step #0 - "yarn": 
> Step #0 - "yarn": 
> Step #0 - "yarn":     'topk' is declared here.
> Step #0 - "yarn":                             ~~~~
> Step #0 - "yarn":     46 export declare const topk: typeof topk_;
> Step #0 - "yarn":   node_modules/@tensorflow/tfjs-core/dist/ops/topk.d.ts:46:22
> Step #0 - "yarn": 
> Step #0 - "yarn":                    ~~~~~~
> Step #0 - "yarn": 278     return tfc.inTopK(yPred, K.flatten(yTrue).asType('float32'), k)
> Step #0 - "yarn": 
> Step #0 - "yarn": src/metrics.ts:278:16 - error TS2551: Property 'inTopK' does not exist on type 'typeof import("/workspace/node_modules/@tensorflow/tfjs-core/dist/index")'. Did you mean 'topk'?
> Step #0 - "yarn": 
> Step #0 - "yarn":     'topk' is declared here.
> Step #0 - "yarn":                             ~~~~
> Step #0 - "yarn":     46 export declare const topk: typeof topk_;
> Step #0 - "yarn":   node_modules/@tensorflow/tfjs-core/dist/ops/topk.d.ts:46:22
> Step #0 - "yarn": 
> Step #0 - "yarn":                    ~~~~~~
> Step #0 - "yarn": 254     return tfc.inTopK(yPred, tfc.argMax(yTrue, -1), k).asType('float32');
> Step #0 - "yarn": 
> Step #0 - "yarn": src/metrics.ts:254:16 - error TS2551: Property 'inTopK' does not exist on type 'typeof import("/workspace/node_modules/@tensorflow/tfjs-core/dist/index")'. Did you mean 'topk'?
> Step #0 - "yarn": $ tsc
> Step #0 - "yarn": [4/4] Building fresh packages...
> Step #0 - "yarn": warning "@tensorflow/tfjs-core > rollup-plugin-visualizer@1.1.1" has incorrect peer dependency "rollup@>=0.60.0".
> Step #0 - "yarn": [3/4] Linking dependencies...
> Step #0 - "yarn": info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
> Step #0 - "yarn": info fsevents@1.2.4: The platform "linux" is incompatible with this module.
> Step #0 - "yarn": [2/4] Fetching packages...
> Step #0 - "yarn": [1/4] Resolving packages...
> Step #0 - "yarn": $ yarn install && yarn build
> Step #0 - "yarn": yarn run v1.13.0
> Step #0 - "yarn": Status: Downloaded newer image for node:10
> Step #0 - "yarn": Digest: sha256:64114d6912ff6fea5af402adcbdb51eeb3541c46a06867510ecd05d171f03dd4
> Step #0 - "yarn": b1efb42c63e2: Pull complete
> Step #0 - "yarn": 6381e1b5feaa: Pull complete
> Step #0 - "yarn": 6381e1b5feaa: Download complete
> Step #0 - "yarn": 6381e1b5feaa: Verifying Checksum
> Step #0 - "yarn": b1efb42c63e2: Download complete
> Step #0 - "yarn": b1efb42c63e2: Verifying Checksum
> Step #0 - "yarn": b1efb42c63e2: Pulling fs layer
> Step #0 - "yarn": 6381e1b5feaa: Pulling fs layer
> Step #0 - "yarn": 6e2490ee2dc8: Already exists
> Step #0 - "yarn": 69d473365bb3: Already exists
> Step #0 - "yarn": d54db43011fd: Already exists
> Step #0 - "yarn": 1b2a72d4e030: Already exists
> Step #0 - "yarn": d4b7902036fe: Already exists
> Step #0 - "yarn": e79bb959ec00: Already exists
> Step #0 - "yarn": 10: Pulling from library/node
> Step #0 - "yarn": Pulling image: node:10
> Starting Step #0 - "yarn"
> BUILD
> Operation completed over 1 objects/402.4 KiB.                                    
> / [0 files][    0.0 B/402.4 KiB]                                                
> / [1 files][402.4 KiB/402.4 KiB]                                                
> Copying gs://834911136599.cloudbuild-source.googleusercontent.com/dfde86cfef887274a7c33249a696d9804b172062-4c81b661-5df9-451d-8703-acc7ef51853b.tar.gz#1557071907252940...
> Fetching storage object: gs://834911136599.cloudbuild-source.googleusercontent.com/dfde86cfef887274a7c33249a696d9804b172062-4c81b661-5df9-451d-8703-acc7ef51853b.tar.gz#1557071907252940
> FETCHSOURCE
> 
> starting build "d9bf7849-4de1-43c0-9349-404fd87dc06f"

**Comment by syt123450** [comment]
Hi @caisq , I also noticed that the cloud build failed. As logging message said `Property 'inTopK' does not exist`, I am afraid the errors were raised by missing `inTopK` op. I added `inTopK` op in this PR [tensorflow/tfjs-core#1734](https://github.com/tensorflow/tfjs-core/pull/1734), and this `topKCategoricalAccuracy  && sparseTopKCategoricalAccuracy`  PR depends on `inTopK` PR. To fix cloud build errors, we may wait for `inTopK` feature to be released in core repository. At that time, we could upgrade tfjs-core version for this repository, and I will merge master into this branch to run cloud build again, I think at that time, all tests would pass.

BTW, could you take a look at the `inTopK` PR here [tensorflow/tfjs-core#1734](https://github.com/tensorflow/tfjs-core/pull/1734)? Thanks~

**Comment by caisq** [comment]
@syt123450 I see: inTopk hasn't even been checked in yet. We need to 1) review and check in that PR in tfjs-core first and 2) wait  for a new version of tfjs-core to be released on NPM.

**Comment by caisq** [comment]
@syt123450 can you take a look at this PR? It's becoming stale and I want to get this checked in soon. We are depending on 1.2.0, which I believe has the necessary new features in tfjs-core.

**Comment by caisq** [comment]
@syt123450 Wait, maybe it's not there yet. Sorry for the confusion.

**Comment by syt123450** [comment]
@caisq Yeah, the inTopK op is still waiting to be reviewed yet.

BTW, I made an update of the jsdoc, removed the usage of `useDocsFrom`. The update makes this PR follow the new jsdoc rule, which is going be to check in in [tensorflow/tfjs-layers#541](https://github.com/tensorflow/tfjs-layers/pull/541).

**Comment by caisq** [comment]
This PR still has build failures even at the latest tfjs-core release (1.2.8). I'm closing the PR now. Please re-open the PR after tfjs-layers is moved to the monorepo at 
https://github.com/tensorflow/tfjs
